src/test/regress/spec/isolation_citus_pause_node.spec

```diff
@@ -0,0 +1,171 @@
+		-- Get a node-id that does not exist in the cluster
+
+		while v_node_exists loop
+			--get a random node id in the range of 1000 to 2000
+			v_node_id :=  FLOOR(RANDOM()*(2000- 1000 + 1)) + 1000;
+			begin
+				select count(0) into v_count from pg_dist_node where nodeid = v_node_id;
+				if v_count = 0 then
+					v_node_exists := false;
+				end if;
+			end;
+		end loop;
+		select citus_pause_node(v_node_id);
+
+
+	EXCEPTION
+		WHEN  SQLSTATE 'P0002' THEN
+			GET STACKED DIAGNOSTICS v_exception_message = MESSAGE_TEXT;
+			v_expected_exception_message := 'node ' || v_node_id || ' not found';
+			if v_exception_message = v_expected_exception_message then
+				RAISE NOTICE 'Node not found.';
+			end if;
+	END;
+	$$
+	LANGUAGE plpgsql;
+}
+
+step "s1-pause-node"
+{
+	SET client_min_messages = 'notice';
+	DO $$
+	DECLARE
+		v_shard_id int;
+		v_node_id int;
+		v_node_name text;
+		v_node_port int;
+	BEGIN
+		--The first message in the block is being printed on the top of the code block. So adding a dummy message
+		--to make sure that the first message is printed in correct place.
+		raise notice '';
+		-- Get the shard id for the distribution column
+		SELECT get_shard_id_for_distribution_column('employee', 3) into v_shard_id;
+
+		--Get the node id for the shard id
+		SELECT nodename,nodeport into v_node_name,v_node_port FROM citus_shards WHERE shardid = v_shard_id limit 1;
+		raise notice 'node name is %',v_node_name;
+		raise notice 'node port is %',v_node_port;
+
+		-- Get the node id for the shard id
+		SELECT nodeid into v_node_id FROM pg_dist_node WHERE nodename = v_node_name and nodeport = v_node_port limit 1;
+
+
+		-- Pause the node
+		perform pg_catalog.citus_pause_node(v_node_id) ;
+	END;
+	$$
+	LANGUAGE plpgsql;
+}
+
+step "s1-end"
+{
+    COMMIT;
+}
+
+session "s2"
+
+
+step "s2-begin"
+{
+	BEGIN;
+}
+
+step "s2-insert"
+{
+	-- Set statement_timeout for the session (in milliseconds)
+	SET statement_timeout = 1000; -- 1 seconds
+	SET client_min_messages = 'notice';
+
+	-- Variable to track if the INSERT statement was successful
+	DO $$
+	DECLARE
+		v_insert_successful BOOLEAN := FALSE;
+	BEGIN
+
+		-- Execute the INSERT statement
+		insert into employee values(11,'e11',3);
+
+		-- If we reach this point, the INSERT statement was successful
+		v_insert_successful := TRUE;
+
+		IF v_insert_successful THEN
+			RAISE NOTICE 'INSERT statement completed successfully. This means that citus_pause_node could not get the lock.';
+		END IF;
+
+
+	-- You can add additional processing here if needed
+	EXCEPTION
+		WHEN query_canceled THEN
+			-- The INSERT statement was canceled due to timeout
+			RAISE NOTICE 'query_canceled exception raised. This means that citus_pause_node was able to get the lock.';
+		WHEN OTHERS THEN
+			-- Any other exception raised during the INSERT statement
+			RAISE;
+	END;
+
+	$$
+	LANGUAGE plpgsql;
+}
+
+step "s2-end"
+{
+	COMMIT;
+}
+
+permutation "s1-begin"  "s1-pause-node" "s2-begin" "s2-insert" "s2-end" "s1-end"
```

Comment:
Can you add more permutations? e.g. 2 concurrent pause_node, pause_node-delete, pause_node-select (even if not blocked). You can also try a concurrent `start_metadata_sync_to_node`.

Reply:
added delete, select. Still trying to figure out start_metadata_sync_to_node

Reply:
@aykut-bozkurt since we are locking the tables in pg_dist_placement, I don't understand why we are testing local tables. Do you want to see that local tables are not being locked?

Antoher one start_metadata_sync_to_node. I don't understand the necessity and scenario of testing this method? Could you clarify more?

Reply:
Yes, Postgres local tables are not in pg_dist_placement (metadata). I mean Citus local tables which are different than Postgres local tables in that we create a single placement at coordinator for them. They can be queried by other nodes (workers) as we store metadata for them. You can check `citus_add_local_table_to_metadata`.

Reply:
I wanted to see if metadata sync should be blocked or not. But I think we do not want to block it to finish the sync. We are only interested in blocking regular user queries. @JelteF Do you agree?

Reply:
@aykut-bozkurt citus_pause_node has a limited functionality to acquire locks. It acquires locks for all the shard placements for a specific node. Since local tables with citus metadata associated are defined only under coordinator, I don't think that these tables are in the scope of this task. @JelteF what do you think?